server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -120,6 +162,23 @@ export class TenantManager {
         return (await this.getTenantDocument(tenantId)).orderer;
     }
 
+    /**
+     * Updates the tenant custom data fields
+     */
+    public async updateCustomData(tenantId: string, customData: ITenantCustomData): Promise<ITenantCustomData> {
+        const db = await this.mongoManager.getDatabase();
+        const collection = db.collection<ITenantDocument>(this.collectionName);
+
+        const customDataUpdateDoc = {};
+        Object.entries(customData).forEach(([key, value]) => {
+            customDataUpdateDoc[`customData.${key}`] = value;
+        });
```

Comment:
Why using a prefixed key? It can be an arbitary object and you can just replace it completely on an update?

Reply:
My thought was that an update to custom data could potentially contain incomplete information. For example, if someone just wants to update tenantAdminContact, but doesn't want to, or know to, update byosAccount info, then it shouldn't wipe the not-included data

Reply:
That does not follow the REST protocol. Its a sub-object and update to it should be atomic.

Reply:
Noted, thanks